packages/styled-components/package.json

```diff
@@ -10,10 +10,33 @@
     "./dist/styled-components.esm.js": "./dist/styled-components.browser.esm.js",
     "./dist/styled-components.cjs.js": "./dist/styled-components.browser.cjs.js"
   },
+  "exports": {
```

Comment:
You are attempting to ship so-called dual package here, thus you are making yourself vulnerable to [the dual package hazard](https://nodejs.org/api/packages.html#dual-package-hazard).  Since styled-components rely on React context, I think that you should try to avoid it as much as possible. Using `.mjs` wrappers strategy would be a much better choice that would be safer for the users.

Reply:
@Andarist I find this very interesting - do you know of any blogpost / resource where the `.mjs` wrappers strategy is explained it detail?

Reply:
It's primarily described in node.js docs [here](https://nodejs.org/api/packages.html#approach-1-use-an-es-module-wrapper) but you can also find references to it in webpack and esbuild docs. We use this strategy in Emotion.

Reply:
ok, thanks. I read that, but got confused with regards to Typescript and the issues described above. I'll have a look on how Emotion handles this as reference for my own lib, thanks!